### PR TITLE
[SPARK-7224] added mock repository generator for --packages tests

### PR DESCRIPTION
**Pull request by brkyvz**
This patch contains an `IvyTestUtils` file, which dynamically generates jars and pom files to test the `--packages` feature without having to rely on the internet, and Maven Central.

cc @pwendell I know that there existed Util functions to create Jars and stuff already, but they didn't really serve my purposes as they appended random prefixes that was breaking things.

I also added the local repository tests. Notice that they work without passing the `repo` to `resolveMavenCoordinates`.

### CODE REVIEW

File: core/src/test/scala/org/apache/spark/deploy/IvyTestUtils.scala [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+  }
+
+  /** Compilable Java Source File */
+  private class JavaSourceFromString(val file: File, val code: String)
+    extends SimpleJavaFileObject(file.toURI, JavaFileObject.Kind.SOURCE) {
+    override def getCharContent(ignoreEncodingErrors: Boolean) = code
+  }
+
+  /** Create a simple testable Class. */
+  private def createJavaClass(dir: File, className: String, packageName: String): File = {
```
Comment (pwendell) [inline core/src/test/scala/org/apache/spark/deploy/IvyTestUtils.scala]: I think this is duplicating some of the logic in:
https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/TestUtils.scala#L114

Comment (brkyvz) [inline core/src/test/scala/org/apache/spark/deploy/IvyTestUtils.scala]: oh yeah, forgot to remove that. Thanks!

File: core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala [hunk 2]
```diff
@@ -731,6 +734,10 @@ private[deploy] object SparkSubmitUtils {
     }
   }
 
+  /** Path of the local Maven cache. */
+  def m2Path: JavaPath = new File(System.getProperty("user.home"),
```
Comment (pwendell) [inline core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala]: make this private spark

**Comment by pwendell** [comment]
Looks good - thanks I can merge this.

**Comment by tgravescs** [comment]
@pwendell @brkyvz   unfortunately java.nio.file.Path (used in core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala)  is a jdk7 only api.  It breaks the build when using jdk6.  Can we revert fix this?

**Comment by tgravescs** [comment]
filed https://issues.apache.org/jira/browse/SPARK-7306

**Comment by pwendell** [comment]
Thanks tom I reverted this.

**Pull request closed without merging**